### PR TITLE
Update Readme.md and fixed Decompress and compress method names

### PR DESCRIPTION
**Pull request by patrickn2**
The idea was just to fix the Readme.md adding gzip compression example but I noticed that the decompress and compress methods were private and then I changed their names to make Decompress and Compress public.

I Also added a gzip test

After this PR I think that is safe to upgrade the package at https://pkg.go.dev/github.com/memcachier/mc/v3

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -30,9 +30,9 @@ import "github.com/memcachier/mc/v3"
 // import "github.com/memcachier/mc"
 
 func main() {
-	// Error handling omitted for demo
+  // Error handling omitted for demo
 
-	// Only PLAIN SASL auth supported right now
+  // Only PLAIN SASL auth supported right now
```
Comment (dterei) [inline README.md:35]: Please fix -- did you swap tabs for space? or vice-versa?

**Review by dterei — CHANGES_REQUESTED** [review]

**Comment by dterei** [comment]
LGTM other than tabs/spaces issue, thanks!

File: README.md (hunk 1 above)
Comment (patrickn2) [reply README.md:35]: Yes I did, before was very bad
Check it out how it is now https://github.com/patrickn2/mc/tree/chore--Update-Readme.md

Comment (patrickn2) [reply README.md:35]: Just saw that is istill bad, Gonna fix asap

Comment (patrickn2) [reply README.md:35]: Fixed

**Pull request merged**